lib/Analysis/Allocation.cpp

```diff
@@ -94,22 +88,33 @@ SmallVector<unsigned> getRepShapeForCvtLayout(RankedTensorType srcTy,
   return repShape;
 }
 
-SmallVector<unsigned>
-getScratchConfigForCvtLayout(triton::gpu::ConvertLayoutOp op, unsigned &inVec,
-                             unsigned &outVec) {
-  auto srcTy = op.getSrc().getType();
-  auto dstTy = op.getType();
-  return getScratchConfigForCvtLayout(srcTy, dstTy, inVec, outVec);
+// TODO: extend beyond scalars
+static SmallVector<unsigned> getRepShapeForAtomicRMW(triton::AtomicRMWOp op) {
+  SmallVector<unsigned> smemShape;
+  if (isa<RankedTensorType>(op.getPtr().getType())) {
+    // do nothing or just assert because shared memory is not used in tensor up
+    // to now
+  } else {
+    // need only bytes for scalar
+    // always vec = 1 and elemsPerThread = 1 for scalar?
+    smemShape.push_back(1);
+  }
+  return smemShape;
+}
+
+static SmallVector<unsigned> getRepShapeForAtomicCAS(triton::AtomicCASOp op) {
+  return SmallVector<unsigned>{1};
 }
 
-SmallVector<unsigned> getScratchConfigForCvtLayout(RankedTensorType srcTy,
-                                                   RankedTensorType dstTy,
-                                                   unsigned &inVec,
-                                                   unsigned &outVec) {
-  auto repShape = getRepShapeForCvtLayout(srcTy, dstTy);
-  if (repShape.empty())
-    return repShape;
-  auto rank = repShape.size();
+ScratchConfig ScratchConfig::get(RankedTensorType srcTy,
+                                 RankedTensorType dstTy) {
+  // Initialize vector sizes and stride
+  ScratchConfig scratchConfig;
+  scratchConfig.repShape = getRepShapeForCvtLayout(srcTy, dstTy);
+  scratchConfig.paddedRepShape = scratchConfig.repShape;
+  if (scratchConfig.repShape.empty())
+    return scratchConfig;
```

Comment:
I think this is returning a ScratchConfig with some fields as uninitialized memory, which I think you do not want.